src/dataprep/main/prep_nsf/scinet_data_to_db.py

```diff
@@ -0,0 +1,97 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# %%
+"""
+Download SciSciNet table SciSciNet_Link_NSF with links between NSF-grants and papers
+Upload into db
+link to Paper_Author_Affiliations, Authors, NSF_Investigator in R file: test_sciscinet_data.R in same folder
+
+
+Create table in database:
+- SciSciNet_Link_NSF
+
+SciSciNet_Link_NSF schema is:
+
+GrantID INTEGER, PaperID INTEGER
+ 
+
+unique index on Grantid and PaperID (multiple paperIDs per GrantID)
+"""
+
+import subprocess
+import sqlite3 as sqlite
+import argparse
+import os
+from os import listdir
+from os.path import isfile, join
+import pandas as pd
+import numpy as np 
+import re 
+import sys
+import requests
+
+sys.path.append('/home/mona/mag_sample/src/dataprep/')  
+
+from helpers.variables import db_file, datapath, databasepath
+from helpers.functions import analyze_db 
+
+scinet_path = os.path.join(datapath, "sciscinet_data/")
+filepath_nsf = os.path.join(scinet_path, "SciSciNet_Link_NSF.tsv")
+
+
+
+# Download file 
+
+url_nsf = "https://ndownloader.figstatic.com/files/36139242"
+response = requests.get(url_nsf)
+with open(filepath_nsf, "wb") as file:
+    file.write(response.content)
+print("Downloaded data")   
+
+
+# ## Read files in loop and dump to db
+
+def load_scinet(filepath):
+    df = pd.read_csv(filepath, 
+                        sep="\t",
+                        names=["NSF_Award_Number", "PaperID", "Type", "Diff_ZScore"])
+    df.drop_duplicates(inplace=True)
+
+    # Create the GrantID column by removing non-numeric characters and formatting
+    
+    df['GrantID'] = df['NSF_Award_Number'].str.extract(r'-(\d+)') 
+    
+    return df
+
+files = [f for f in listdir(scinet_path) if isfile(join(scinet_path, f))]
+
+
+con = sqlite.connect(database = db_file, isolation_level= None)
+with con: 
+    for (i,f) in enumerate(files):
+        df = load_scinet(scinet_path+f)
+        #print(df.head())
+        if i==0:
+            if_exists_opt="replace"
+        else:
+            if_exists_opt="append"
+
+        df.to_sql("scinet", 
```

Comment:
I would prefer a more descriptive if longer table name, ie "scinet_links", or something. @chrished ?

It looks like there's something wrong the the data are stored in the database at the moment:
```bash
NSF_Award_Number|PaperID|Type|Diff_ZScore|GrantID
NSF_Award_Number|PaperID|Type|Diff_ZScore| #this row should not be here
NSF-1907207|3101155693|First||1907207
NSF-1900929|2966372995|First||1900929
NSF-1900929|2973880421|First||1900929
NSF-1900929|3042450767|First||1900929
NSF-1900929|2987739614|First||1900929
```

Also, do we need both GrantID and NSF_Award_Number?

Reply:
Okay, I'll adjust the name. 
I wanted to keep both variables for now to be sure that there's no inconsistencies between the raw data set and the data set I use  to link with mag authors, but I can of course remove the NSF_Award_Number

Reply:
Should I then also remove the Diff_ZScore variable before loading it to the db?

Reply:
>Should I then also remove the Diff_ZScore variable before loading it to the db?
See christoph's comment below. 

> be sure that there's no inconsistencies between the raw data set and the data set I use
That's a good idea (what's the exact check you want to run?). Perhaps you can do this check separately (depending on how extensive it is, in a separate file or not), before writing the data to the database?

Reply:
For now I just looked at the number of rows to be sure that no observations are lost and when creating an index it could serve as a check for unique values at the same time